src/envoy/tcp/mixer/filter.cc

```diff
@@ -211,6 +211,9 @@ bool Filter::GetDestinationIpPort(std::string *str_ip, int *port) const {
       filter_callbacks_->upstreamHost()->address()) {
     return Utils::GetIpPort(filter_callbacks_->upstreamHost()->address()->ip(),
                             str_ip, port);
+  } else {
```

Comment:
Hmm, how did this work before for server side reports? Is this being overridden somewhere later?
I'm a bit worried since this changes the telemetry for client-side reports for destination IP.

Reply:
So the change only matters when there's no assigned upstream cluster which gets triggered for BlackHoleCluster or other clusters which are not defined or clusters with no endpoints. For every other case the `if` clause should get invoked and preserve old behavior.

I think this is a net improvement on the existing behavior as it ensures that we always have a DestinationIP where as in the existing implementation there will be scenarios where destination IP and port will be "unknown" or "0".

Reply:
I think the cluster is not assigned at the moment of the call here. The upstream pool sends a callback after the connection is allowed (or perhaps, this is a race). If that's the case, then there is a change in the behavior.

Reply:
I see. If my understanding is correct then for BlackHole a cluster will never get assigned as without this logic I was never getting Destination IP in my telemetry.

For other cases, you're right this might be a race. I will add some logging and verify if that's the case.

Reply:
This is only called in onLog, so the decision from the upstream pool should have already been made by then.